pkg/common/gatewayapi_utils.go

```diff
@@ -10,6 +10,7 @@ import (
 	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	gatewayapiv1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
```

Comment:
why keep using v1alpha2?

Reply:
Not everything seems to have moved to v1beta1, so i just stuck with the minimal changes to make it work with the latest version.

Reply:
It looks like `PolicyTargetReference` is now the only type we use that still requires v1alpha2.

Reply:
I believe so yes